CMakeLists.txt

```diff
@@ -1,45 +1,50 @@
 cmake_minimum_required(VERSION 3.10)
+project(server017 VERSION 0.0.1 DESCRIPTION "Server of team 17 for the Sopra")
 
 # build options
-set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-
 if (NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE Release)
 endif (NOT CMAKE_BUILD_TYPE)
 
-set(CMAKE_CXX_FLAGS "-Wall -Wextra -Wpedantic -Werror -mtune=native -march=native")
+set(COMMON_CXX_FLAGS -Wall -Wextra -Wpedantic -Werror -mtune=native -march=native)
 if (CMAKE_BUILD_TYPE STREQUAL "Debug")
     message("\n##########################")
     message("###### DEBUG BUILD #######")
     message("##########################\n")
+    list(APPEND COMMON_CXX_FLAGS -Og --coverage)
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} --coverage")
 else ()
     message("\n############################")
     message("###### RELEASE BUILD #######")
     message("############################\n")
+    list(APPEND COMMON_CXX_FLAGS -O3)
 endif ()
 
-project(server017 VERSION 0.0.1 DESCRIPTION "Server of team 17 for the Sopra")
-
 # Libraries
 # spdlog
 find_package(spdlog REQUIRED)
-SET(LIBS ${LIBS} spdlog::spdlog)
+list(APPEND LIBS spdlog::spdlog)
 
 # CLI11
 find_package(CLI11 REQUIRED)
 add_definitions(-DCLI11_HAS_FILESYSTEM=1)
-SET(LIBS ${LIBS} CLI11::CLI11 stdc++fs)
+list(APPEND LIBS CLI11::CLI11 stdc++fs)
 
 # LibCommon
-set(LIBCOMMON_TESTS OFF CACHE BOOL "Build LibCommon tests" FORCE)
 add_subdirectory(external/LibCommon)
-SET(LIBS ${LIBS} SopraCommon)
+list(APPEND LIBS SopraCommon)
+
+# WebsocketCPP
+add_subdirectory(external/WebsocketCPP)
+list(APPEND LIBS SopraNetwork)
 
-file(GLOB_RECURSE SOURCES
-        ${CMAKE_SOURCE_DIR}/src/*.cpp
-        ${CMAKE_SOURCE_DIR}/src/*.hpp)
+set(SOURCES
+        src/main.cpp)
```

Comment:
Was hälst du davon hier dann auch die includes explicit aufzulisten?

Reply:
Finde ich nicht notwendig, in den Sources haben wir das bisher noch nie gemacht... Das hatten wir mal im LibClient für das install target (jetzt auch nicht mehr...)

Die benötigten header werden ja in den sources inkludiert...